### PR TITLE
DashboardScene serialization: Handle variables and library panels 

### PR DESCRIPTION
**Pull request by dprokop**
Variables and library panels serializatio, required for snapshots (not only of course...(

### CODE REVIEW

File: public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts [hunk 1]
```diff
@@ -245,3 +279,100 @@ export function dataLayersToAnnotations(layers: SceneDataLayerProvider[]) {
...
+          // @ts-expect-error
+          value: variable.state.value,
+        },
+        options: variable.state.options.map((option) => ({
+          text: option.label,
+          value: option.value.toString(),
+          selected: Array.isArray(variable.state.value)
+            ? Boolean(variable.state.value.find((v) => v === option.value))
+            : variable.state.value === option.value,
+        })),
```
Comment (torkelo) [inline public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts:333]: this dos not look correct, for custom variables only the query property should be persisted not the options

Comment (dprokop) [reply public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts:333]: I was as surprised as you are :)

File: public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts [hunk 2]
```diff
@@ -245,3 +279,100 @@ export function dataLayersToAnnotations(layers: SceneDataLayerProvider[]) {
...
+        multi: variable.state.isMulti,
+        skipUrlSync: Boolean(variable.state.skipUrlSync),
+        hide: variable.state.hide || VariableHide.dontHide,
+      });
+    } else if (variable instanceof CustomVariable) {
+      variables.push({
+        ...commonProperties,
+        current: {
+          // TODO
+          // selected: false
```
Comment (torkelo) [inline public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts:321]: there is nothing todo, we do not need to handle selected prop.

**Review by torkelo — COMMENTED** [review]
Looks good, only a minor issue with CustomVariable, we can remove the options from the save model, they should not be there (I know they are in the old model but this could be seen as a bug)

**Review by torkelo — APPROVED** [review]

**Pull request merged**